File: web/src/app/services/ride-analytics.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { RideAnalyticsResponse } from '../models/ride-analytics.model';

@Injectable({ providedIn: 'root' })
export class RideAnalyticsService {
  private baseUrl = 'http://localhost:8081/api';

  constructor(private http: HttpClient) {}

  getDailyAggregates(start: string, end: string, role?: string, userId?: string, aggregate?: boolean): Observable<RideAnalyticsResponse> {
    let params = new HttpParams().set('start', start).set('end', end);

    if (role) params = params.set('role', role);
    if (userId) params = params.set('userId', userId);
    if (aggregate) params = params.set('aggregate', String(aggregate));

    return this.http.get<RideAnalyticsResponse>(`${this.baseUrl}/analytics/rides`, { params });
  }
}

File: mobile/app/src/main/java/com/example/gotaximobile/network/RideAnalyticsService.java
package com.example.gotaximobile.network;

import com.example.gotaximobile.models.dtos.RideAnalyticsResponseDTO;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface RideAnalyticsService {

    @GET("api/analytics/rides")
    Call<RideAnalyticsResponseDTO> getRideAnalytics(
            @Query("start") String start,
            @Query("end") String end,
            @Query("role") String role,
            @Query("userId") String userId,
            @Query("aggregate") boolean aggregate
    );
}
